### PR TITLE
cpu/stm32f3: added SPI driver

### PR DESCRIPTION
**Pull request by haukepetersen**

### CODE REVIEW

File: cpu/stm32f3/periph/spi.c [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+    int i, trans_ret, trans_bytes = 0;
+    char in_temp;
+
+    for (i = 0; i < length; i++) {
+        if (out != NULL) {
+            trans_ret = spi_transfer_byte(dev, out[i], &in_temp);
+        }
+        else {
+            trans_ret = spi_transfer_byte(dev, NULL, &in_temp);
```
Comment (PeterKietzmann) [inline cpu/stm32f3/periph/spi.c]: This leads to the warning `expected 'char' but argument is of type 'void *' `

**Comment by PeterKietzmann** [comment]
Except from the above mentioned warning and the needed squashing this PR seems to be nearly ready to merge

**Comment by haukepetersen** [comment]
squashed and fixed source of warning

**Comment by haukepetersen** [comment]
and rebased, of course...

**Comment by PeterKietzmann** [comment]
Travis: "Pull request needs squashing according to its labels"

**Comment by haukepetersen** [comment]
should be fine now?!

**Comment by thomaseichinger** [comment]
ACK & Go

**Pull request merged**